[docs] Standardize getting started docs across all packages (#16302)
mui/mui-x: docs/data/charts/quickstart/quickstart.md

- Utilities – such as classes and types

See the [Charts composition documentation](/x/react-charts/composition/) for complete details.

The demo below shows how to use composition to create a custom Chart that combines a Bar and a Line Chart on a single plot:

{{"demo": "Combining.js"}}

### Chart layouts

The layout of a Chart is defined by two main spaces: the plot area, and the outer margins.

The `width` and `height` props define the dimensions of the SVG which is the root of the chart.
Within this SVG, the plot area (or drawing area) serves as the canvas for data visualization, where the lines, bars, or other visual elements are rendered.
The size of the plot area is determined by the `margin = {top, bottom, left, right}` object which defines its outer margins inside the SVG.
The outer margin space is where information like axes, titles, and legends are displayed.

See the [Styling documentation](/x/react-charts/styling/#placement) for complete details.

### Server-side rendering

Charts support server-side rendering under two conditions:

1. `width` and `height` props must be provided – it's not possible to compute the SVG dimensions on the server.

> 2. Animations must be disabled with the `skipAnimation` prop – otherwise the animation will be in an empty state on first render.

Reviewer: To avoid the "will" but not sure
```suggestion
2. Animations must be disabled with the `skipAnimation` prop – otherwise the first render show the initial frame of the animation.
```

  },
});
```

## Using this documentation

Most Chart components have two accompanying documents:

1. **Overview** – a general description of built-in features

> 2. **Demo** – a collection of custom examples

Author: Good catch!

> 2. **Demo** – a collection of custom examples

Author: Good catch!

> 2. **Demo** – a collection of custom examples

Author: ```suggestion
2. **Demos** – a collection of custom examples
```

> 2. **Demos** – a collection of custom examples

Reviewer 2: Not sure how necessary this section is, as it can be deduced from reading the sidebar

Author: True, probably don't really need to spell this out. "Using this documentation" is a helpful section for the other products but maybe less useful here. Unless there's anything else you can think of that the reader should know about the docs before continuing past this page?

Reviewer 2: I can't think of any extra info the user should `need` to know.
